Mtd.Kiosk.Annunciator.Readers.Raspi/PiReader.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Mtd.Kiosk.Annunciator.Core;
+using Mtd.Kiosk.Annunciator.Readers.Raspi.Config;
+using System.Device.Gpio;
+
+namespace Mtd.Kiosk.Annunciator.Readers.Raspi;
+
+public sealed class PiReader : IButtonReader, IDisposable
+{
+    private readonly ILogger<PiReader> _logger;
+
+    private readonly int[] _gpioPins;
+    private bool disposedValue;
+
+    private GpioController? _controller;
+
+    public event EventHandler? ButtonPressed;
+
+    public string Name => "Raspberry Pi Reader";
+
+    public PiReader(IOptions<PiReaderConfig> options, ILogger<PiReader> logger)
+    {
+        ArgumentNullException.ThrowIfNull(options?.Value, nameof(options.Value));
+        ArgumentNullException.ThrowIfNull(logger, nameof(logger));
+        ArgumentOutOfRangeException.ThrowIfLessThanOrEqual(options.Value.Pins.Length, 0, nameof(options.Value.Pins));
+
+        for (var pin = 0; pin < options.Value.Pins.Length; pin++)
+        {
+            ArgumentOutOfRangeException.ThrowIfLessThanOrEqual(options.Value.Pins[pin], 0, nameof(options.Value.Pins));
+        }
+
+        _logger = logger;
+
+        _gpioPins = options.Value.Pins;
+
+        _controller = new GpioController();
+    }
+
+    private void Callback(object sender, PinValueChangedEventArgs args)
+    {
+        ButtonPressed?.Invoke(this, EventArgs.Empty);
```

Comment:
This seems like a good place to log the pin and change type.

```suggestion
		_logger.LogDebug("Button {changeType} on pin {pin}", args.ChangeType, args.PinNumber);
        ButtonPressed?.Invoke(this, EventArgs.Empty);
```